### ISSUE TITLE
PacBio reads with uniformly high quality scores are read as Solexa encoding.

### ISSUE THREAD

**Issue by brendanf** [benjjneb/dada2#682]
Apparently `ShortRead` recognizes [two standards for encoding quality scores](https://rdrr.io/bioc/ShortRead/man/readFastq.html#heading-3) in fastq: `FastqQuality`, a.k.a. "Phred-like", where the value is ASCII - 33, and `SFastqQuality`, a.k.a. "Solexa-like", where the value is ASCII - 64.  The default behavior is to search the quality scores given for the first 10000 bases in the file.  If all are >=58 and some are greater than 74, then `SFastqQuality` is assumed; otherwise `FastqQuality`.

I guess this is reasonable for Illumina or IonTorrent data, but by the time I have run `filterAndTrim()` on a 1000-bp amplicon in PacBio, filtering out all sequences with more than 3 expected errors, then I actually have a lot of reads where all bases are quality > 25 (ASCII 58), and the median score is quality 93 (ascii 126).  Apparently, in at least one of my files this is true for the first 10000 bases (~10 reads), and then they are being incorrectly read using `SFastqQuality`.  Of course it is a problem to have wrong quality scores, but it is worse than that, because when a base has an average quality scores < 31 (ASCII 64), then this results in negative quality scores, which cause an error in `dada()`:
```
Invalid derep$quals matrix. Quality values must be positive integers.
```

The most straight-forward solution would be to add a `qualityType` (or `...`) argument to `derepFastq()` which is then passed to `yield()`.  This might also be a good idea for other functions which use `readFastq()` or `FastqStreamer`, although this is the place where I think it is most likely to cause trouble, because most earlier steps will be dealing with files which include some bases with quality < 25.

I will test the change and send a pull request if it seems to work.

**Comment by benjjneb** [comment]
Thanks for this report, I had not run into this yet but it certainly makes sense. I'd be happy to consider a pull request that addresses this issue. 

If you don't mind, I might bring this up to the Bioconductor folks that maintain ShortRead as well, as I think we are the main package right now using them to process PacBio CCS reads.

**Comment by benjjneb** [comment]
Fixed in f96fdc6cf51596d8007e9a28bc13d73ef4d2193d
